### ISSUE TITLE
Need better error message when using Auto mode on Linux and no ssh-agent is running and no socket file is specified in the configuration.

### ISSUE THREAD

**Issue by dachande** [dlech/KeeAgent#166]
I've recently updated KeeAgent to the latest version 0.8.1-1 and KeePass2 has previously been updated to 2.34

Since the update of KeeAgent, it won't start anymore with the following message:
`Object reference not set to instance of an object`

I'm using Arch Linux and Linux Mint and I noticed this behaviour on both machines.

**Comment by dlech** [comment]
Do you get a more informative message if you run `keepass2 --debug` from a terminal?

**Comment by dachande** [comment]
The only console output I receive when running keepass with the `--debug` parameter is:
`SendMessage (14680106, 0x112c, 0x4, 0x4)`

**Comment by dlech** [comment]
Do you have any other plugins installed?

**Comment by dachande** [comment]
Yes, I am using `keepass-plugin-http 1.8.4.1-1`
I temporarily removed that plugin and cleared the plugin cache but that didn't help.

**Comment by dachande** [comment]
It looks like I have fixed the problem.

I was running KeeAgent in Auto Mode. As no ssh-agent was running, KeeAgent started in Agent Mode and maybe printed out this error message because no socket file was specified in the configuration. As I switched KeeAgent to Client Mode and set up ssh-agent correctly, the error message disappeard and KeeAgent is now successfully sending the keys to ssh-agent for authorization.

**Comment by dlech** [comment]
I'm re-opening since we should have a better error message when this happens. I'm glad you were able to track down the problem.

**Comment by dachande** [comment]
I've just experimented further on my other machine using KeeAgent in Agent Mode. The error message is indeed related to this issue as it disappeard as soon as I switched KeeAgent to Agent Mode and specified a socket file. I think that it will work as well in Auto Mode as long as a socket file is specified.

I noticed that after switching to Agent Mode and specifying a socket file another error message popped up stating that the unix socket cannot be opened. I think that's because the socket file is opened by KeeAgent on startup (after a restart) of Keepass. Maybe this one should be fixed as well, to tell the user, that a restart of Keepass is required to apply the changes.
